core/src/Kokkos_Macros.hpp

```diff
@@ -172,7 +172,10 @@
 
 #define KOKKOS_IMPL_HIP_CLANG_WORKAROUND
 
+#define HIP_ENABLE_PRINTF
+#define HCC_ENABLE_PRINTF
```

Comment:
I don't think this does anything. Like you said you still need to export the environment variable.